fastapi/_compat.py

```diff
@@ -109,9 +110,24 @@ def type_(self) -> Any:
             return self.field_info.annotation
 
         def __post_init__(self) -> None:
-            self._type_adapter: TypeAdapter[Any] = TypeAdapter(
-                Annotated[self.field_info.annotation, self.field_info]
-            )
+            with warnings.catch_warnings():
+                # Pydantic >= 2.12.0a1 warns about this when building
+                # TypeAdapters from field information that uses aliases.
+                # The Pydantic team recommends ignoring this in this case:
+                # https://github.com/fastapi/fastapi/pull/14036#issuecomment-3316045587
+                try:
+                    from pydantic.warnings import (  # type: ignore[attr-defined]
+                        UnsupportedFieldAttributeWarning,
+                    )
+                except ImportError:  # pragma: no cover
+                    pass
+                else:  # pragma: no cover
+                    warnings.simplefilter(
+                        "ignore", category=UnsupportedFieldAttributeWarning
+                    )
+                self._type_adapter: TypeAdapter[Any] = TypeAdapter(
+                    Annotated[self.field_info.annotation, self.field_info]
+                )
```

Comment:
- I believe you need to use the `catch_warnings` context manager, because as it stands the warning will always be ignored from now on.
- I would suggest using an explicit version check on the `pydantic` version instead of a `try..except` statement, so that it can easily be unconditionally imported once FastAPI drops support for Pydantic < 2.12.

Reply:
I'm already using the `catch_warnings` context manager just above, so I'm confused about your first point.

I've made your suggested second change.